src/SFA.DAS.EAS.Web/Orchestrators/CommitmentStatusCalculator.cs

```diff
@@ -53,5 +56,32 @@ private RequestStatus GetEmployerOnlyStatus(LastAction lastAction, bool hasAppre
 
             return RequestStatus.None;
         }
+
```

Comment:
I not sure whether we should throw an exception here or not! There's very good arguments to do so, but this user will not likely be able to fix the problem. But I suppose they will call us

Reply:
I suggest we want to fail fast as soon as we know something's gone wrong. It makes it easier to track down bugs.
If we didn't throw an exception, we'd either have to return a bogus status which would cause an insidious bug further down the line, or we'd have to return an error code, which the consumer could ignore and again lead to a harder to track down bug.

Reply:
From https://en.wikipedia.org/wiki/Fail-fast...

* Fail-fast components are often used in situations where failure in one component might not be visible until it leads to failure in another component.
* Finding the cause of a failure is easier in a fail-fast system, because the system reports the failure with as much information as possible as close to the time of failure as possible. In a fault-tolerant system, the failure might go undetected, whereas in a system that is neither fault-tolerant nor fail-fast the failure might be temporarily hidden until it causes some seemingly unrelated problem later.
* A fail-fast system that is designed to halt as well as report the error on failure is less likely to erroneously perform an irreversible or costly operation.